File: cms-app-crawler-py/src/pages/global_element.py
from ..base.base_page import BasePage
from ..base.locator import Locator


class GlobalElement(BasePage):
    _MEDICARE_NAV = Locator("css", 'a[href*="medicare"]')
    _MEDICAID_NAV = Locator("css", 'a[href*="medicaid"]')
    _ABOUT_CMS    = Locator("css", 'a[href*="about-cms"]')
    _EMAIL_SIGNUP = Locator("css", 'input[type="email"]')

    async def click_medicare_nav(self) -> "GlobalElement":
        await self.click(self._MEDICARE_NAV)
        return self

    async def click_medicaid_nav(self) -> "GlobalElement":
        await self.click(self._MEDICAID_NAV)
        return self

    async def click_about_cms(self) -> "GlobalElement":
        await self.click(self._ABOUT_CMS)
        return self

File: cms-app-crawler-js/src/pages/global/GlobalElement.js
'use strict';

const { BasePage } = require('../../base/BasePage');
const { Locator } = require('../../base/Locator');

class GlobalElement extends BasePage {
  static MEDICARE_NAV   = new Locator('css', 'a[href*="medicare"]');
  static MEDICAID_NAV   = new Locator('css', 'a[href*="medicaid"]');
  static ABOUT_CMS      = new Locator('css', 'a[href*="about-cms"]');
  static EMAIL_SIGNUP   = new Locator('css', 'input[type="email"]');

  async clickMedicareNav() {
    await this.click(GlobalElement.MEDICARE_NAV);
    return this;
  }

  async clickMedicaidNav() {
    await this.click(GlobalElement.MEDICAID_NAV);
    return this;
  }

  async clickAboutCms() {
    await this.click(GlobalElement.ABOUT_CMS);
    return this;
  }
}

module.exports = { GlobalElement };
